libvast/src/data.cpp

```diff
@@ -197,7 +199,14 @@ caf::error unpack(const fbs::Data& from, data& to) {
       return caf::none;
     }
     case fbs::data::Data::pattern: {
-      to = pattern{from.data_as_pattern()->value()->str()};
+      auto options = pattern_options{};
+      options.case_insensitive
+        = from.data_as_pattern()->options()->case_insensitive();
```

Comment:
This will crash when reading an old pattern that did not have this field set because you dereference a nullptr. Please never blindly do so, but rather put an assertion to indicate that you think as a dev that it cannot be a nullptr, or, well, check for it.

Reply:
Ooooh, I didn't realize `flatbuffers` tends to hand out dumb pointers in this way. 
I added a `nullptr` check in https://github.com/tenzir/vast/pull/2964/commits/cee68e11879e936384dd048c2a2121cf83788325.